nexus/src/app/affinity.rs

```diff
@@ -0,0 +1,407 @@
+        &self,
+        opctx: &OpContext,
+        project_lookup: &lookup::Project<'_>,
+        pagparams: &PaginatedBy<'_>,
+    ) -> ListResultVec<views::AffinityGroup> {
+        let (.., authz_project) =
+            project_lookup.lookup_for(authz::Action::ListChildren).await?;
+
+        Ok(self
+            .db_datastore
+            .affinity_group_list(opctx, &authz_project, pagparams)
+            .await?
+            .into_iter()
+            .map(Into::into)
+            .collect())
+    }
+
+    pub(crate) async fn anti_affinity_group_list(
+        &self,
+        opctx: &OpContext,
+        project_lookup: &lookup::Project<'_>,
+        pagparams: &PaginatedBy<'_>,
+    ) -> ListResultVec<views::AntiAffinityGroup> {
+        let (.., authz_project) =
+            project_lookup.lookup_for(authz::Action::ListChildren).await?;
+
+        Ok(self
+            .db_datastore
+            .anti_affinity_group_list(opctx, &authz_project, pagparams)
+            .await?
+            .into_iter()
+            .map(Into::into)
+            .collect())
+    }
+
+    pub(crate) async fn affinity_group_create(
+        &self,
+        opctx: &OpContext,
+        project_lookup: &lookup::Project<'_>,
+        affinity_group_params: params::AffinityGroupCreate,
+    ) -> CreateResult<views::AffinityGroup> {
+        let (.., authz_project) =
+            project_lookup.lookup_for(authz::Action::CreateChild).await?;
+
+        let affinity_group =
+            AffinityGroup::new(authz_project.id(), affinity_group_params);
+        self.db_datastore
+            .affinity_group_create(opctx, &authz_project, affinity_group)
+            .await
+            .map(Into::into)
+    }
+
+    pub(crate) async fn anti_affinity_group_create(
+        &self,
+        opctx: &OpContext,
+        project_lookup: &lookup::Project<'_>,
+        anti_affinity_group_params: params::AntiAffinityGroupCreate,
+    ) -> CreateResult<views::AntiAffinityGroup> {
+        let (.., authz_project) =
+            project_lookup.lookup_for(authz::Action::CreateChild).await?;
+
+        let anti_affinity_group = AntiAffinityGroup::new(
+            authz_project.id(),
+            anti_affinity_group_params,
+        );
+        self.db_datastore
+            .anti_affinity_group_create(
+                opctx,
+                &authz_project,
+                anti_affinity_group,
+            )
+            .await
+            .map(Into::into)
+    }
+
+    pub(crate) async fn affinity_group_update(
+        &self,
+        opctx: &OpContext,
+        group_lookup: &lookup::AffinityGroup<'_>,
+        updates: &params::AffinityGroupUpdate,
+    ) -> UpdateResult<views::AffinityGroup> {
+        let (.., authz_group) =
+            group_lookup.lookup_for(authz::Action::Modify).await?;
+        self.db_datastore
+            .affinity_group_update(opctx, &authz_group, updates.clone().into())
+            .await
+            .map(|g| g.into())
```

Comment:
huh, can this not be
```suggestion
            .map(Into::into)
```